CK3toEU4/Source/CK3World/Religions/Faiths.cpp

```diff
@@ -31,6 +31,14 @@ void CK3::Faiths::linkReligions(const Religions& religions)
 		if (religionDataItr != religionData.end())
 		{
 			faith.second->loadReligion(*religionDataItr);
+			for (const auto& title: titles.getTitles())
```

Comment:
Iterating over all faiths and then iterating over all titles smells of a double loop that could be sped up by caching beforehand. For example, before the double loop you could run a titles loop once and cache all title ids, and then in the faith loop all you have to look up is if the religious head is in the map (.contains()). Much faster, generally.